oceannavigator/frontend/src/components/DatasetSelector.jsx

```diff
@@ -108,19 +113,19 @@ function DatasetSelector({
         // Carry the currently selected variable to the new
         // dataset if said variable exists in the new dataset.
         setLoadingPercent(33);
-        let newVariable = currentVariable;
-        let newVariableScale = newVarScale ?? mountedDataset.variable_scale;
-        let newQuiver = newQuiverVariable ?? mountedDataset.quiverVariable;
+        newVariable = newVariable ?? mountedDataset.variable
+        newVariableScale = newVariableScale ?? mountedDataset.variable_scale;
```

Comment:
Prevents scale from updating when a dataset is changed without having a variable selected.